helpers/integration.go

```diff
@@ -236,6 +236,9 @@ func CalculateIntegrationPipelineRunOutcome(adapterClient client.Client, ctx con
 		if err != nil {
 			return false, fmt.Errorf("error while getting test results from pipelineRun %s: %w", pipelineRun.Name, err)
 		}
+	} else if HasPipelineRunFinished(pipelineRun) && !HasPipelineRunSucceeded(pipelineRun) {
```

Comment:
Thinking about this some more, having this condition in the `else if` block means that a failed pipeline with one passing task would be considered as passed overall. 
I think it might be better to first check if the pipeline succeeded and then check the task child references (meaning swapping [this condition](https://github.com/redhat-appstudio/integration-service/pull/166/files#diff-ea27f7cada77e890e9c4d991d8edc8f9c6bfbbf2a830cf74df4ba606341240caR239) with  [this one](https://github.com/redhat-appstudio/integration-service/pull/166/files#diff-ea27f7cada77e890e9c4d991d8edc8f9c6bfbbf2a830cf74df4ba606341240caR233).

Reply:
Thinking about this some more, having this condition in the else if block means that a failed pipeline with one passing task would be considered as passed overall.                        => This situation was not considered before and

Reply:
Thanks!